crates/typst/src/engine.rs

```diff
@@ -40,6 +46,48 @@ impl Engine<'_> {
             }
         }
     }
+
+    fn get_lang_bundle<'a>(
+        &'a mut self,
+        lang_id: &str,
+    ) -> &'a FluentBundle<FluentResource> {
+        if !self.langs.contains_key(lang_id) {
+            let id = FileId::new(None, VirtualPath::new(lang_id.to_string() + ".ftl"));
+            let fluent_file =
+                self.world.file(id).expect("no valid fluent file in project root");
+            let ftl_string =
+                String::from_utf8(fluent_file.to_vec()).expect("no valid utf8");
+
+            let res = FluentResource::try_new(ftl_string)
+                .expect("Failed to parse an FTL string.");
+            let langid_en: LanguageIdentifier = "en-US".parse().expect("Parsing failed");
+            let mut bundle = FluentBundle::new(vec![langid_en]);
+            bundle
+                .add_resource(res)
+                .expect("Failed to add FTL resources to the bundle.");
```

Comment:
Afaik all error messages in Typst are lower case. For consistency, you should probably do the same.

I'm also wondering whether such specific message should really be exposed to the user? Maybe just a generic "failed to load translations for specified language" would be enough? Not sure though.

Reply:
Once the resources are embeded in the binary (which they probably should be, as mentioned in the issue) this will / should be update to only possible throw a build time error, not a run time one anyway. For now this is only developer facing not user facing because it isn't a production arrangement yet (as I understood the PR state).

Reply:
Also this is an incorrect use of `expect`. `expect` isn't an assertion, see https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style

Reply:
How is this incorrect? Isn't `.expect()` basically just "try to unwrap() and if success move on, otherwise throw this text string as an error if failed"?

Reply:
What I think they are saying is: It is correct to call `expect` here, but the wording of its argument is incorrect. It should be `expect("there needs to be a valid fluent file in the project root")` instead of `expect("no valid fluent file in project root")`.
I admit, I just copied the code from the example in https://docs.rs/fluent/0.16.0/fluent/ to have something working.

Reply:
The next thing I'm gonna do is migrating all languages to ftl files and including them into the binary.